cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -341,13 +354,20 @@ impl<State> CoreEnvironment<State> {
             out_link_path.as_ref().display()
         );
 
+        let service_config_path = if let LockedManifest::Catalog(ref lockfile) = lockfile {
+            maybe_make_service_config_file(flox, lockfile)?
+        } else {
+            None
+        };
+
```

Comment:
nonblocking greenkeeping: afaict the only difference between `build` and `link` is that `link` passes through two args to `lockfile.build`, so I think we should factor out the shared logic rather than adding the same logic in multiple places

Reply:
It looks like:
- `build` does "always build and never link"
- `link` does "maybe build and always link"

The combination of which would be something like `maybe_build_and_maybe_link` and I'm not sure the extra abstraction makes it any clearer. It would be nice if build/link were two separate operations/paths but they're buried in `pkgdb` which `LockedManifest` is wrapping. Any suggestions?

Reply:
> It would be nice if build/link were two separate operations/paths 

💯

fwiw link should basically be `nix build --out-link <path> <store path>` and not do any of the _actual building_ (nix build is guarateed to not build anything iff `<store path>` already exists).

We basically do the same in pkgdb, but ensure the lined path exists by buidling it, insofar it is more like `nix build --out-link <path> <flake installable or .nix-file>` today

Reply:
From sync: I'll timebox some effort on Monday, separate to this PR, to look into changing our use of `link` so that it doesn't optionally build.